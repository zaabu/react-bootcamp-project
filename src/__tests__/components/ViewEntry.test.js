import React from "react";
import { shallow, configure } from "enzyme";
import configureMockStore from "redux-mock-store";
import ViewEntry from "../../components/entries/viewDiary";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

describe("SignupUser component", () => {
  const props = {
    dispatch: jest.fn(),
    buttonLabel: "",
    className: "",
    name: "",
    purpose: "",
    due_date: "",
    date_created: ""
  };
  const mockStore = configureMockStore();
  let wrapper;

  beforeEach(() => {
    const store = mockStore({});
    wrapper = shallow(<ViewEntry {...props} store={store} />);
  });
  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  
});
