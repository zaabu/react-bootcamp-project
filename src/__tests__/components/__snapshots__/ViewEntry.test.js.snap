// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupUser component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ViewEntry
    buttonLabel=""
    className=""
    date_created=""
    dispatch={[MockFunction]}
    due_date=""
    name=""
    purpose=""
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <button
          class="btn btn-success"
          id="success-button"
          onClick={[Function]}
          type="button"
        >
          View
        </button>,
        <Modal
          autoFocus={true}
          backdrop={true}
          backdropTransition={
            Object {
              "mountOnEnter": true,
              "timeout": 150,
            }
          }
          centered={false}
          className=""
          fade={true}
          isOpen={false}
          keyboard={true}
          modalTransition={
            Object {
              "timeout": 300,
            }
          }
          onClosed={[Function]}
          onOpened={[Function]}
          role="dialog"
          toggle={[Function]}
          zIndex={1050}
        >
          <ModalHeader
            charCode={215}
            close={
              <button
                className="close"
                onClick={[Function]}
              >
                ×
              </button>
            }
            closeAriaLabel="Close"
            tag="h5"
            toggle={[Function]}
            wrapTag="div"
          />
          <ModalBody
            tag="div"
          />
          <ModalFooter
            tag="div"
          >
            <Badge
              className="float-left"
              color="warning"
              pill={false}
              tag="span"
            >
              Date Created - 
              
            </Badge>
             
            <Badge
              className="float-left"
              color="info"
              pill={false}
              tag="span"
            >
              Last Modified - 
            </Badge>
          </ModalFooter>
        </Modal>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "View",
          "class": "btn btn-success",
          "id": "success-button",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "View",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoFocus": true,
          "backdrop": true,
          "backdropTransition": Object {
            "mountOnEnter": true,
            "timeout": 150,
          },
          "centered": false,
          "children": Array [
            <ModalHeader
              charCode={215}
              close={
                <button
                  className="close"
                  onClick={[Function]}
                >
                  ×
                </button>
              }
              closeAriaLabel="Close"
              tag="h5"
              toggle={[Function]}
              wrapTag="div"
            />,
            <ModalBody
              tag="div"
            />,
            <ModalFooter
              tag="div"
            >
              <Badge
                className="float-left"
                color="warning"
                pill={false}
                tag="span"
              >
                Date Created - 
                
              </Badge>
               
              <Badge
                className="float-left"
                color="info"
                pill={false}
                tag="span"
              >
                Last Modified - 
              </Badge>
            </ModalFooter>,
          ],
          "className": "",
          "fade": true,
          "isOpen": false,
          "keyboard": true,
          "modalTransition": Object {
            "timeout": 300,
          },
          "onClosed": [Function],
          "onOpened": [Function],
          "role": "dialog",
          "toggle": [Function],
          "zIndex": 1050,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "charCode": 215,
              "children": undefined,
              "close": <button
                className="close"
                onClick={[Function]}
              >
                ×
              </button>,
              "closeAriaLabel": "Close",
              "tag": "h5",
              "toggle": [Function],
              "wrapTag": "div",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": undefined,
              "tag": "div",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Badge
                  className="float-left"
                  color="warning"
                  pill={false}
                  tag="span"
                >
                  Date Created - 
                  
                </Badge>,
                " ",
                <Badge
                  className="float-left"
                  color="info"
                  pill={false}
                  tag="span"
                >
                  Last Modified - 
                </Badge>,
              ],
              "tag": "div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    "Date Created - ",
                    "",
                  ],
                  "className": "float-left",
                  "color": "warning",
                  "pill": false,
                  "tag": "span",
                },
                "ref": null,
                "rendered": Array [
                  "Date Created - ",
                  "",
                ],
                "type": [Function],
              },
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    "Last Modified - ",
                    undefined,
                  ],
                  "className": "float-left",
                  "color": "info",
                  "pill": false,
                  "tag": "span",
                },
                "ref": null,
                "rendered": Array [
                  "Last Modified - ",
                  undefined,
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <button
            class="btn btn-success"
            id="success-button"
            onClick={[Function]}
            type="button"
          >
            View
          </button>,
          <Modal
            autoFocus={true}
            backdrop={true}
            backdropTransition={
              Object {
                "mountOnEnter": true,
                "timeout": 150,
              }
            }
            centered={false}
            className=""
            fade={true}
            isOpen={false}
            keyboard={true}
            modalTransition={
              Object {
                "timeout": 300,
              }
            }
            onClosed={[Function]}
            onOpened={[Function]}
            role="dialog"
            toggle={[Function]}
            zIndex={1050}
          >
            <ModalHeader
              charCode={215}
              close={
                <button
                  className="close"
                  onClick={[Function]}
                >
                  ×
                </button>
              }
              closeAriaLabel="Close"
              tag="h5"
              toggle={[Function]}
              wrapTag="div"
            />
            <ModalBody
              tag="div"
            />
            <ModalFooter
              tag="div"
            >
              <Badge
                className="float-left"
                color="warning"
                pill={false}
                tag="span"
              >
                Date Created - 
                
              </Badge>
               
              <Badge
                className="float-left"
                color="info"
                pill={false}
                tag="span"
              >
                Last Modified - 
              </Badge>
            </ModalFooter>
          </Modal>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "View",
            "class": "btn btn-success",
            "id": "success-button",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "View",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoFocus": true,
            "backdrop": true,
            "backdropTransition": Object {
              "mountOnEnter": true,
              "timeout": 150,
            },
            "centered": false,
            "children": Array [
              <ModalHeader
                charCode={215}
                close={
                  <button
                    className="close"
                    onClick={[Function]}
                  >
                    ×
                  </button>
                }
                closeAriaLabel="Close"
                tag="h5"
                toggle={[Function]}
                wrapTag="div"
              />,
              <ModalBody
                tag="div"
              />,
              <ModalFooter
                tag="div"
              >
                <Badge
                  className="float-left"
                  color="warning"
                  pill={false}
                  tag="span"
                >
                  Date Created - 
                  
                </Badge>
                 
                <Badge
                  className="float-left"
                  color="info"
                  pill={false}
                  tag="span"
                >
                  Last Modified - 
                </Badge>
              </ModalFooter>,
            ],
            "className": "",
            "fade": true,
            "isOpen": false,
            "keyboard": true,
            "modalTransition": Object {
              "timeout": 300,
            },
            "onClosed": [Function],
            "onOpened": [Function],
            "role": "dialog",
            "toggle": [Function],
            "zIndex": 1050,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "charCode": 215,
                "children": undefined,
                "close": <button
                  className="close"
                  onClick={[Function]}
                >
                  ×
                </button>,
                "closeAriaLabel": "Close",
                "tag": "h5",
                "toggle": [Function],
                "wrapTag": "div",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": undefined,
                "tag": "div",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <Badge
                    className="float-left"
                    color="warning"
                    pill={false}
                    tag="span"
                  >
                    Date Created - 
                    
                  </Badge>,
                  " ",
                  <Badge
                    className="float-left"
                    color="info"
                    pill={false}
                    tag="span"
                  >
                    Last Modified - 
                  </Badge>,
                ],
                "tag": "div",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      "Date Created - ",
                      "",
                    ],
                    "className": "float-left",
                    "color": "warning",
                    "pill": false,
                    "tag": "span",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Date Created - ",
                    "",
                  ],
                  "type": [Function],
                },
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      "Last Modified - ",
                      undefined,
                    ],
                    "className": "float-left",
                    "color": "info",
                    "pill": false,
                    "tag": "span",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Last Modified - ",
                    undefined,
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
